runtime/src/iree/builtins/ukernel/BUILD.bazel

```diff
@@ -14,6 +14,10 @@ package(
     licenses = ["notice"],  # Apache 2.0
 )
 
+# iree_bitcode_library generates targets on the other package
+# we need to expose files used by it.
+exports_files(glob(["*.h"]))
```

Comment:
Can you clarify what this fixes? (e.g. share some logs of what failed to build)

I thought some of this code might need to move files around or export files, but I'm wondering why our Bazel CI is happy with what we have if you evidently ran into some issues... maybe a difference between Bazel and Blaze?

Reply:
I didn't know about `exports_files`, but reading the documentation now,
https://bazel.build/reference/be/functions#exports_files
it makes sense and the implicit-exporting behavior that we were relying on here is indeed the kind of thing that Bazel would frown upon as sloppy.

The above doc link mentions that `--incompatible_no_implicit_file_export` gives stricter behavior. I'm trying now locally to check that it does cause Bazel to reject the current code and that this PR fixes that, and I'll report back.

Reply:
Yeah let's turn on `--incompatible_no_implicit_file_export` upstream and here

Reply:
Filed https://github.com/openxla/iree/issues/14020 about that.